### ISSUE TITLE
Different default values of learning rate  lead to different results

### ISSUE THREAD

**Issue by Neronjust2017** [keras-team/autokeras#1000]
Hi,
 When I first ran my experiment with autokeras (v1.0.0) , in the fuction **compile_keras_model** in the file **autokeras/hypermodel/graph.py**, I set the default value of learning rate to **0.01**, as you can see from the code below:

```
def _compile_keras_model(self, hp, model):
        # Specify hyperparameters from compile(...)
        optimizer_name = hp.Choice('optimizer',
                              ['adam', 'adadelta', 'sgd'],
                              default='adam')
        optimizer = keras.optimizers.get(optimizer_name)
        optimizer.learning_rate = hp.Choice(
            'learning_rate', [0.1, 0.01, 0.001], default=0.01)
        model.compile(optimizer=optimizer,
                      metrics=self._get_metrics(),
                      loss=self._get_loss())
        return model
```

As a result, I got the 10 best trials,  they all have a learning rate of **0.01**, and a optimizer of **adam**. Importantly, they all have a relatively high **Score**, which means  the best nerual architectures found in this run have a  relatively high **val_loss**, as you can see below:

_Score: 0.1710327418986708     learning_rate: 0.01  optimizer: adam
Score: 0.18145588133484125    learning_rate: 0.01  optimizer: adam
Score: 0.196065416559577        learning_rate: 0.01  optimizer: adam
Score: 0.197494484949857        learning_rate: 0.01  optimizer: adam
Score: 0.19837430003099144    learning_rate: 0.01  optimizer: adam
Score: 0.19994546892121434    learning_rate: 0.01  optimizer: adam
Score: 0.20179287670180202    learning_rate: 0.01  optimizer: adam
Score: 0.20277987327426672    learning_rate: 0.01  optimizer: adam
Score:  0.20279210386797786   learning_rate: 0.01  optimizer: adam
Score:  0.20463723689317703   learning_rate: 0.01  optimizer: adam_

After fully trained, the metrics on best model:
_loss: 0.20353382487188687
accuracy: 0.88834953
precision: 0.8910985
recall: 0.8910985
f1: 0.8910985_

And when I ran for the second time, I changed the code above:

```
 def _compile_keras_model(self, hp, model):
        # Specify hyperparameters from compile(...)
        optimizer = hp.Choice('optimizer',
                              ['adam', 'adadelta', 'sgd'],
                              default='adam')
        model.compile(optimizer=optimizer,
                      metrics=self._get_metrics(),
                      loss=self._get_loss())
        return model
```

Also, I got the 10 best trials,  all of them have a optimizer of **adam**, and a learning rate of **0.001**(because the default  learning rate of Adam is 0.001), and they all have a relatively low **Score**, which means  the best nerual architectures found in this run have a  relatively low **val_loss**.

_Score: 0.05167456396156922  optimizer: adam
Score: 0.0734095184598118     optimizer: adam
Score: 0.07529105222783983   optimizer: adam
Score: 0.07614397187717259   optimizer: adam
Score: 0.07943510916084051   optimizer: adam
Score: 0.08357392949983478   optimizer: adam
Score: 0.09189658565446734   optimizer: adam
Score: 0.11087238369509578   optimizer: adam
Score: 0.11638897564262152   optimizer: adam
Score: 0.12073865439742804   optimizer: adam_

After fully trained, the metrics on best model:
_loss: 0.16954380915514333      
accuracy: 0.96407765 
precision: 0.9649621                
recall: 0.9649621             
f1: 0.9649621_

And when I changed the code as below, I got a similar result — almost all of them have a optimizer of **adam**, and a learning rate of **0.001**.

```
def _compile_keras_model(self, hp, model):
        # Specify hyperparameters from compile(...)
        optimizer_name = hp.Choice('optimizer',
                              ['adam', 'adadelta', 'sgd'],
                              default='adam')
        optimizer = keras.optimizers.get(optimizer_name)
        optimizer.learning_rate = hp.Choice(
            'learning_rate', [0.1, 0.01, 0.001], default=0.001)
        model.compile(optimizer=optimizer,
                      metrics=self._get_metrics(),
                      loss=self._get_loss())
        return model
```

By the way, I used Greedy tuner. And it seems that it tends to search near the default value, even though there are other better structures defined in the search space. What do you think is the reason for that?
Thanks.

**Comment by haifeng-jin** [comment]
10 trials is still too few for the tuner to explore the search space. That is why the learning rate didn't change during search. Greedy is tend to explore around the tried hyper parameters. We believe that would be a good strategy for hyper parameter tuning.

We are still in a stage of make autokeras 1.0 robust. Haven't fine tune the tuning algorithms yet.
